### ISSUE TITLE
Complex datatype support?

### ISSUE THREAD

**Issue by sunilkpai** [NVIDIA/AMGX#104]
In the documentation, it says there is complex datatype support that is currently limited. To what extent is complex supported by AMGX for Krylov iterative solver methods and how can I specify this in the config?

Thanks for your help!

**Comment by asluckybirdsa** [comment]
I have the same question. It seems that type hZZI can be used to perform complex computation. However, the complier always fails when the system solver is built.

**Comment by marsaev** [comment]
Tested configuration was GMRES + Aggregation Multigrid + Jacobi solver. This is still an experimental feature and actually didn't get much traction. We don't have specific date where we will get back to this.

Mode is specified during solver creation - see `mode` parameter in `AMGX_solver_create()`. See `AMGX_Mode` enum in `amgx_config.h` for possible values. After created solver will expect only specific vector and matrix datatypes only defined by this `mode`

**Comment by marsaev** [comment]
> However, the complier always fails when the system solver is built.

Right, sometimes some code changes are required to enable complex support for specific component (i.e. solver).

**Comment by marsaev** [comment]
Added task to provide complex support matrix for algorithms to documentation
